lib/linux/dmi_sysfs.go

```diff
@@ -60,36 +61,56 @@ func DMIInfoFromSysfs() (*DMIInfo, error) {
 // The method reads as much information as possible, so it always returns a
 // non-nil [DMIInfo], even if it errors.
 func DMIInfoFromFS(dmifs fs.FS) (*DMIInfo, error) {
-	var vals []string
+	var wg sync.WaitGroup
+
+	var mu sync.Mutex // guards errs
```

Comment:
Up to you, but I think this would be cleaner without the mutex. You could:

- allocate `errs` with the correct capacity ahead of time
- update the loop: `for i, spec := range`, then each goroutine writes to `errors[i]`, which is not racy
- run `errors.Join` after the `wg.Wait`, which will discard any nil errors

Reply:
Good call, I don't like having >1 concurrency control variable at the same time but this didn't occur to me then.

Changed to rely more on indexes, so did away with specs for a list of names too. Looks a bit more like the original.